arch/x86/core/ia32/intstub.S

```diff
@@ -80,13 +84,13 @@ SECTION_FUNC(TEXT, _interrupt_enter)
 	 */
 
 #ifdef CONFIG_EXECUTION_BENCHMARKING
-	pushl %eax
-	pushl %edx
-	rdtsc
-	mov %eax, arch_timing_irq_start
-	mov %edx, arch_timing_irq_start+4
-	pop %edx
-	pop %eax
+	push	%eax
+	push	%edx
+
+	call	read_timer_start_of_isr
```

Comment:
this is wrong, you have to push ALL caller-saved registers if you are calling into C code since we are at the beginning of the ISR code where nothing has been preserved yet.
the old code pushed only eax/edx because of how 'rdtsc' works.